### PR TITLE
add decimal sort and decimal compare2

### PR DESCRIPTION
**Pull request by broccoliSpicy**
**What type of PR is this?**

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

**Which issue(s) this PR fixes:**

issue #2463 

**What this PR does / why we need it:**

Not Available

**Special notes for your reviewer:**

Not Available

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

### CODE REVIEW

File: pkg/compare2/desc/decimal128s/compare.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
+package decimal128s
```
Comment (yingfeng) [inline pkg/compare2/desc/decimal128s/compare.go:1]: License required

File: pkg/compare2/desc/decimal128s/compare_test.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
+package decimal128s
```
Comment (yingfeng) [inline pkg/compare2/desc/decimal128s/compare_test.go:1]: License required

File: pkg/compare2/desc/decimal128s/types.go [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package decimal128s
```
Comment (yingfeng) [inline pkg/compare2/desc/decimal128s/types.go:1]: License required

File: pkg/compare2/desc/decimal64s/compare.go [hunk 4]
```diff
@@ -0,0 +1,41 @@
+package decimal64s
```
Comment (yingfeng) [inline pkg/compare2/desc/decimal64s/compare.go:1]: License required

File: pkg/compare2/desc/decimal64s/compare_test.go [hunk 5]
```diff
@@ -0,0 +1,57 @@
+package decimal64s
```
Comment (yingfeng) [inline pkg/compare2/desc/decimal64s/compare_test.go:1]: License required

File: pkg/compare2/desc/decimal64s/types.go [hunk 6]
```diff
@@ -0,0 +1,13 @@
+package decimal64s
```
Comment (yingfeng) [inline pkg/compare2/desc/decimal64s/types.go:1]: License required

**Review by nnsgmsone — COMMENTED** [review]
lgtm

**Pull request merged**